src/main/java/cat/udl/eps/entsoftarch/textannot/domain/Annotation.java

```diff
@@ -32,12 +32,15 @@
     @ManyToOne
     private Sample annotated;
 
+    @ManyToOne
+    private Tag tagged;
+
     public Annotation(){
 
     }
 
     @Override
     public Integer getId() {
-        return id;
+        return null;
```

Comment:
Why getId() returns always null?
BTW, if you use lombok's @DaTa , as id is a non-final instance variable, you get getId. And, if you're using lombok, you can use @NoArgsConstructor for lombok to create one (if you need it).